packages/cli/src/pages/api/messages/index.ts

```diff
@@ -15,7 +17,94 @@ export default async function handler(
     return res.status(405).json({ error: "Method not allowed" });
   }
 
-  if (!(await validateApiKey(req, res))) return;
+  const apiKey = apiKeyFromReq(req);
+
+  if (typeof apiKey !== "string") {
+    res.status(422).end("expected x-api-key in header or apiKey in query");
+    return false;
+  }
+
+  let organizationId;
+
+  try {
+    const apiKeyRecord = await prisma.apiKey.findFirstOrThrow({
+      where: {
+        id: apiKey,
+        active: true,
+      },
+    });
+
+    organizationId = apiKeyRecord.organizationId;
+  } catch {
+    res.status(401).end("API key is not valid");
+  }
+
+  const organization = await prisma.organization.findFirstOrThrow({
+    where: { id: organizationId },
+  });
+
+  const listId = req.query.listId || req.body.listId;
+  let list, defaultList, listMember, defaultListMember;
+
+  // if listId exists, look it up
+  if (listId) {
+    list = await prisma.list.findFirstOrThrow({ where: { id: listId } });
+  }
+
+  // also lookup the default list
+  if (list?.isDefault) {
+    defaultList = list;
+  } else {
+    defaultList = await prisma.list.findFirstOrThrow({
+      where: { organizationId: organization.id, isDefault: true },
+    });
+  }
+
+  if (list) {
+    listMember = await prisma.member.findFirst({
+      where: { listId: list.id, email: req.body.email },
+    });
+  }
+
+  if (list !== defaultList) {
+    defaultListMember = await prisma.member.findFirst({
+      where: { listId: defaultList.id, email: req.body.email },
+    });
+  }
+
+  // return early if your status to either list is not "subscribed"
+  if (
+    listMember?.status !== "subscribed" ||
+    defaultListMember?.status !== "subscribed"
+  ) {
+    res.status(200).json({ message: "user is not subscribed to either list" });
+  }
```

Comment:
- needs a different condition, should be false if user is not a member of the list
- should return
- message is supposed to be a model, use error

```suggestion
  if (
    listMember?.status === "unsubscribed" ||
    defaultListMember?.status === "unsubscribed"
  ) {
    res.status(200).json({ error: "user is not subscribed to either list" });
    return;
  }
```